### ISSUE TITLE
`CA2014: Do not use stackalloc in loops` should not be report for local function inside loop

### ISSUE THREAD

**Issue by yyjdelete** [dotnet/roslyn-analyzers#4135]
### Analyzer

**Diagnostic ID**: [CA2014](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2014?view=vs-2019): `CA2014: Do not use stackalloc in loops`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 5.0.100-preview.8](https://dotnet.microsoft.com/download/dotnet/5.0)

### Describe the bug

`CA2014: Do not use stackalloc in loops` should not be report for local function inside loop

### Steps To Reproduce

```cs
        public static void XXX()
        {
            while (true)
            {
                XX();

                static void XX()
                {
                    Span<int> tmp = stackalloc int[10];
                    Console.WriteLine(tmp[0]);
                }
            }
        }
```

### Expected behavior
No CA2014, since it the same as the below code.
```cs
        public static void XXX()
        {
            while (true)
            {
                XX();
            }

                static void XX()
                {
                    Span<int> tmp = stackalloc int[10];
                    Console.WriteLine(tmp[0]);
                }
        }
```

### Actual behavior
CA2014

**Comment by mavasani** [comment]
@Evangelink @Youssef1313 interested in picking this up? This analyzer is warning by default in the.NET5 SDK, so high priority. Please post a comment if you want to work on it, otherwise I can work on a fix tomorrow.

**Comment by Youssef1313** [comment]
@mavasani Thanks for the ping. I opened a PR and will add a test to it now.
